src/sql_types.rs

```diff
@@ -89,104 +90,125 @@ pub struct Enum {
 }
 
-#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq)]
+#[derive(Deserialize, Clone, Debug, Eq, PartialEq)]
 pub struct Index {
-    pub oid: u32,
     pub table_oid: u32,
-    pub name: String,
-    pub column_attnums: Vec<i32>,
+    pub column_names: Vec<String>,
     pub is_unique: bool,
     pub is_primary_key: bool,
-    pub comment: Option<String>,
-}
-
-#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq)]
-pub struct ForeignKeyPermissions {
-    // Are tables + columns on both sides of the fkey selectable?
-    pub is_selectable: bool,
 }
 
-#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq)]
+#[derive(Deserialize, Clone, Debug, Eq, PartialEq)]
 pub struct ForeignKeyTableInfo {
     pub oid: u32,
     // The table's actual name
     pub name: String,
-    pub column_attnums: Vec<i32>,
+    pub schema: String,
     pub column_names: Vec<String>,
-    pub directives: TableDirectives,
 }
 
-#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq)]
+#[derive(Deserialize, Clone, Debug, Eq, PartialEq)]
 pub struct ForeignKeyDirectives {
-    pub inflect_names: bool,
     pub local_name: Option<String>,
     pub foreign_name: Option<String>,
 }
 
-#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq)]
+#[derive(Deserialize, Clone, Debug, Eq, PartialEq)]
 pub struct ForeignKey {
-    pub oid: u32,
-    pub name: String,
-    pub is_locally_unique: bool,
-    pub permissions: ForeignKeyPermissions,
     pub directives: ForeignKeyDirectives,
     pub local_table_meta: ForeignKeyTableInfo,
     pub referenced_table_meta: ForeignKeyTableInfo,
 }
 
-#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq)]
+#[derive(Deserialize, Clone, Debug, Eq, PartialEq)]
 pub struct TableDirectiveTotalCount {
     pub enabled: bool,
 }
 
-#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq)]
+#[derive(Deserialize, Clone, Debug, Eq, PartialEq)]
 pub struct TableDirectives {
     pub inflect_names: bool,
     pub name: Option<String>,
     // XXX: comment directive key is totalCount
     pub total_count: Option<TableDirectiveTotalCount>,
+
+    // Views / Materialized Views only:
+    // @graphql({"primary_key_columns": ["id"]})
+    pub primary_key_columns: Option<Vec<String>>,
 }
 
-#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq)]
+#[derive(Deserialize, Clone, Debug, Eq, PartialEq)]
 pub struct Table {
     pub oid: u32,
     pub name: String,
     pub schema: String,
     pub columns: Vec<Arc<Column>>,
     pub comment: Option<String>,
+    pub relkind: String, // r = table, v = view, m = mat view, f = foreign table
     pub permissions: TablePermissions,
     pub indexes: Vec<Index>,
     pub functions: Vec<Arc<Function>>,
-    pub foreign_keys: Vec<Arc<ForeignKey>>,
     pub directives: TableDirectives,
 }
 
 impl Table {
-    pub fn primary_key(&self) -> Option<&Index> {
-        self.indexes.iter().find(|x| x.is_primary_key)
+    pub fn primary_key(&self) -> Option<Index> {
```

Comment:
extended to out primary key defined in a comment directive if no native primary key is found